src/shared_modules/content_manager/tests/unit/APIDownloader_test.hpp

```diff
@@ -22,6 +26,85 @@ class APIDownloaderTest : public ::testing::Test
+
+    std::shared_ptr<UpdaterBaseContext> m_spUpdaterBaseContext; ///< UpdaterBaseContext used on the merge pipeline.
+
+    std::shared_ptr<APIDownloader> m_spAPIDownloader; ///< APIDownloader used to download the content.
+
+    inline static std::unique_ptr<FakeServer> fakeServer; ///< pointer to FakeServer class
+
+    /**
+     * @brief Sets initial conditions for each test case.
+     *
+     */
+    // cppcheck-suppress unusedFunction
+    void SetUp() override
+    {
+        m_spAPIDownloader = std::make_shared<APIDownloader>();
+        // Create a updater base context
+        m_spUpdaterBaseContext = std::make_shared<UpdaterBaseContext>();
+        m_spUpdaterBaseContext->outputFolder = "/tmp/api-downloader-tests";
+        m_spUpdaterBaseContext->downloadsFolder = m_spUpdaterBaseContext->outputFolder / DOWNLOAD_FOLDER;
+        m_spUpdaterBaseContext->contentsFolder = m_spUpdaterBaseContext->outputFolder / CONTENTS_FOLDER;
+        m_spUpdaterBaseContext->configData = R"(
+            {
+                "contentSource": "api",
+                "compressionType": "raw",
+                "versionedContent": "false",
+                "deleteDownloadedContent": false,
+                "url": "http://localhost:4444/raw",
+                "outputFolder": "/tmp/api-downloader-tests",
+                "dataFormat": "json",
+                "contentFileName": "sample.json"
+            }
+        )"_json;
+        // Create a updater context
+        m_spUpdaterContext = std::make_shared<UpdaterContext>();
+        // Create folders
+        std::filesystem::create_directory(m_spUpdaterBaseContext->outputFolder);
+        std::filesystem::create_directory(m_spUpdaterBaseContext->downloadsFolder);
+        std::filesystem::create_directory(m_spUpdaterBaseContext->contentsFolder);
+    }
+
+    /**
+     * @brief Tear down routine for tests
+     *
+     */
+    // cppcheck-suppress unusedFunction
+    void TearDown() override
+    {
+        // Remove outputFolder
+        std::filesystem::remove_all(m_spUpdaterBaseContext->outputFolder);
+        // Reset APIDownloader
+        m_spAPIDownloader.reset();
+        // Reset UpdaterContext
+        m_spUpdaterContext.reset();
+        // Reset UpdaterBaseContext
+        m_spUpdaterBaseContext.reset();
+    }
+
+    /**
+     * @brief Creates the fakeServer for the runtime of the test suite
+     */
+    // cppcheck-suppress unusedFunction
+    static void SetUpTestSuite()
+    {
+        if (!fakeServer)
+        {
+            fakeServer = std::make_unique<FakeServer>("localhost", 4444);
+        }
+    }
+
+    /**
+     * @brief Resets fakeServer causing the shutdown of the test server.
+     */
+    // cppcheck-suppress unusedFunction
+    static void TearDownTestSuite()
+    {
+        fakeServer.reset();
+    }
 };
 
-#endif //_API_DOWNLOADER_TEST_HPP
+#endif //_API_DOWNLOADER_TEST_HPP
```

Comment:
Missing new line at the end of file:
```suggestion
#endif //_API_DOWNLOADER_TEST_HPP

```